packages/test/test-version-utils/src/compatConfig.ts

```diff
@@ -130,6 +131,49 @@ const genLTSConfig = (compatVersion: number | string): CompatConfig[] => {
 	];
 };
 
+const genBackCompatConfig = (compatVersion: number): CompatConfig[] => {
+	return [
+		{
+			name: `compat back N${compatVersion} - older loader`,
+			kind: CompatKind.Loader,
+			compatVersion,
+			loader: compatVersion,
+		},
+		{
+			name: `compat back N${compatVersion} - older loader + older driver`,
+			kind: CompatKind.LoaderDriver,
+			compatVersion,
+			driver: compatVersion,
+			loader: compatVersion,
+		},
+	];
+};
+
+const genFullBackCompatConfig = (): CompatConfig[] => {
+	const _configList: CompatConfig[] = [];
+	// This will need to be updated once we move beyond 2.0.0-internal.x.y.z
+	let semverInternal: string | undefined;
+	if (pkgVersion.startsWith("2.0.0-internal.")) {
+		semverInternal = pkgVersion.split("internal.")[1];
+	}
+
+	if (pkgVersion.startsWith("2.0.0-dev.")) {
+		semverInternal = pkgVersion.split("dev.")[1];
+	}
+
+	assert(semverInternal !== undefined, "Unexpected pkg version");
+	// This is to make pipeline runs work
+	semverInternal = semverInternal.split(".").slice(0, 3).join(".");
+
+	const semverVal = new semver.SemVer(semverInternal);
+	const num = semverVal.major;
+	// This makes the assumption N and N-1 scenarios are already fully tested thus skipping 0 and -1.
+	for (let i = 2; i < num; i++) {
```

Comment:
So what would `num` be in this case? How far back are we looking? 

To put things into perspective, the original bug was between `2.0.0-internal.1.4.6` and `2.0.0-internal.2.2.0`.